src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -273,109 +273,99 @@ private static ChatRole ToChatRole(global::Azure.AI.Inference.ChatRole role) =>
-            result.MaxTokens = options.MaxOutputTokens;
-            result.NucleusSamplingFactor = options.TopP;
-            result.PresencePenalty = options.PresencePenalty;
-            result.Temperature = options.Temperature;
-            result.Seed = options.Seed;
-
-            if (options.StopSequences is { Count: > 0 } stopSequences)
-            {
-                foreach (string stopSequence in stopSequences)
-                {
-                    result.StopSequences.Add(stopSequence);
-                }
-            }
+            result = azureAIOptions;
+            result.Messages = ToAzureAIInferenceChatMessages(chatContents).ToList();
+            result.Model = options.ModelId ?? _metadata.DefaultModelId ?? throw new InvalidOperationException("No model id was provided when either constructing the client or in the chat options.");
+        }
+        else
+        {
+            result = CreateAzureAIOptions(chatContents, options);
+        }
+
+        result.FrequencyPenalty = options.FrequencyPenalty;
+        result.MaxTokens = options.MaxOutputTokens;
+        result.NucleusSamplingFactor = options.TopP;
+        result.PresencePenalty = options.PresencePenalty;
+        result.Temperature = options.Temperature;
+        result.Seed = options.Seed;
 
-            // These properties are strongly typed on ChatOptions but not on ChatCompletionsOptions.
-            if (options.TopK is int topK)
+        if (options.StopSequences is { Count: > 0 } stopSequences)
+        {
+            foreach (string stopSequence in stopSequences)
             {
-                result.AdditionalProperties["top_k"] = new BinaryData(JsonSerializer.SerializeToUtf8Bytes(topK, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(int))));
+                result.StopSequences.Add(stopSequence);
             }
+        }
 
-            if (options.AdditionalProperties is { } props)
+        if (options.Tools is { Count: > 0 } tools)
+        {
+            foreach (AITool tool in tools)
             {
-                foreach (var prop in props)
+                if (tool is AIFunction af)
                 {
-                    switch (prop.Key)
-                    {
-                        // Propagate everything else to the ChatCompletionsOptions' AdditionalProperties.
-                        default:
-                            if (prop.Value is not null)
-                            {
-                                byte[] data = JsonSerializer.SerializeToUtf8Bytes(prop.Value, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(object)));
-                                result.AdditionalProperties[prop.Key] = new BinaryData(data);
-                            }
-
-                            break;
-                    }
+                    result.Tools.Add(ToAzureAIChatTool(af));
                 }
             }
 
-            if (options.Tools is { Count: > 0 } tools)
+            switch (options.ToolMode)
```

Comment:
Does result.ToolChoice default to null? Assuming yes, we should probably only do this switch if result.ToolChoice is null, so as not to overwrite an explicit setting the user providing the raw representation already made.

If, however, result.ToolChoice defaults to non-null, we should stick with what you have.